test/template-tests/testcases/deploymentTemplate/ManagedIdentityExtension-must-not-be-used.test.ps1

```diff
@@ -1,11 +1,11 @@
 ﻿param(
 [Parameter(Mandatory=$true,Position=0)]
 [string]
-$TemplateText
+$TemplateObject
 )
-if ($templateText -match 'managedidentityextension') {
-    Write-Error "Managed Identity Extension must not be used" -ErrorId ManagedIdentityExtension.Was.Used
-}
-
 
+$resourcesJson = $TemplateObject.resources  | ConvertTo-Json -Depth 10  

```

Comment:
1) Let's build some "well known global vars" for the parameters, variables and resources... 
2) while building resources, let's flatten the child resources in the hashtable.
3) update all the resource tests to use this new well known vars we create